src/lang-items.md

```diff
@@ -0,0 +1,289 @@
+# Lang items
+
+
+The `rustc` compiler has certain pluggable operations, that is,
+functionality that isn't hard-coded into the language, but is
+implemented in libraries, with a special marker to tell the compiler
+it exists. The marker is the attribute `#[lang = "..."]` and there are
+various different values of `...`, i.e. various different 'lang
+items'.
+
+For example, `Box` pointers require two lang items, one for allocation
+and one for deallocation. A freestanding program that uses the `Box`
+sugar for dynamic allocations via `malloc` and `free`:
```

Comment:
Hmm, I'm not sure this is a great example because you can do this more easily with `#[global_allocator]`. Maybe use `#[lang = "start"]` instead? I think you can use that to have a graphical entrypoint instead of a console entrypoint on Windows. It would also simplify the example a lot I think - this shouldn't be a section on how to write an allocator ;)